### ISSUE TITLE
registerOrRemoveHandler: the target element for event handler registration does not exist, when processing the following event handler registration:

### ISSUE THREAD

**Issue by parisseb** [MoAlyousef/fltk_wasm32_emscripten#4]
I see several error messages of that kind for emscripten_set_mousedown_callback and alike. My guess is that this happens at window creation, probably in makeWindow, because the window itself is not yet registered in the DOM when the emscripten_set_##ev##_callback is called. Do you observe the same ?
I do not understand really the issue. Maybe most of the time, when the first emscripten callback is called, the window exists, and everything is fine, but if not then this might freeze the app. Any idea how to modify the registration process?

**Comment by parisseb** [comment]
More info: the issue is inside delete_div.
I have a window created makewindow #fltk_canvas1 but immediatly after delete_div #fltk_canvas1668509029, with apparently a random window id.

**Comment by MoAlyousef** [comment]
I'm not able see this. 
The register macros run in the makeWindow function, so the window should already be registered in the DOM. 

In `Fl_Emscripten_Window_Driver::hide()` the global fl_window is set to 0. Maybe I should add a check that if fl_em_find returns NULL, it should return early from delete_div. However the xid is only taken from an existing window, so that should never fail.

Do you have some reproduction code?

**Comment by parisseb** [comment]
It's a relatively large app, you can get the source here:
https://www-fourier.univ-grenoble-alpes.fr/~parisse/giac/emgiac3.tgz
Compile in emgiac3/giac by running make xcas.html after editing the Makefile, in LDJSFLAGS remove -L/Users/parisse/emfltk/lib (this is the location where I have a binary emscripten build of the fltk lib so that I can copy it on another PC easily)
The issue happens when you click on a menu for example, in the browser console you will see error messages. My guess is that there might be some windows that is already created by a new, but was never shown (so the canvas id would be somewhat random), and perhaps some code tries to hide it. My current workaround is a check in fl_em_find return a null pointer when xid is too large (2^48 right now, this is of course not a real fix, but it should be good enough for running the app a few days).

**Comment by MoAlyousef** [comment]
Ah yes I now understand what you mean. This occurs because of a limitation in emscripten:

https://emscripten.org/docs/porting/asyncify.html#reentrancy

I describe it briefly here in the 1st limitation:
https://github.com/MoAlyousef/fltk_wasm32_emscripten/blob/emscripten/README.Emscripten.txt

I also have asked about any workarounds in the emscripten forum:
https://groups.google.com/g/emscripten-discuss/c/NS-IAsPA51

As of now, I don’t know of a way to work around this issue. 

Note that Qt works around it by not using emscripten_sleep in the event loop, as such menus don’t work in such cases.

**Comment by parisseb** [comment]
Yes, I'm aware of the fact that Fl::wait is not reentrant and I need to remove creation of modal windows in callbacks from buttons in my app. From menus it seems however safe (and menus are themselves creating window). And the issue happens when I open a menu.
